src/spatialdata/_core/query/spatial_query.py

```diff
@@ -102,6 +104,165 @@ def _get_bounding_box_corners_in_intrinsic_coordinates(
     return intrinsic_bounding_box_corners, intrinsic_axes
 
 
+def _get_polygon_in_intrinsic_coordinates(
+    element: DaskDataFrame | GeoDataFrame, target_coordinate_system: str, polygon: Polygon | MultiPolygon
+) -> GeoDataFrame:
+    from spatialdata._core.operations.transform import transform
+
+    polygon_gdf = ShapesModel.parse(GeoDataFrame(geometry=[polygon]))
+
+    m_without_c, input_axes_without_c, output_axes_without_c = _get_axes_of_tranformation(
+        element, target_coordinate_system
+    )
+    spatial_transform = Affine(m_without_c, input_axes=input_axes_without_c, output_axes=output_axes_without_c)
+
+    m_without_c_linear = m_without_c[:-1, :-1]
+    case = _get_case_of_bounding_box_query(m_without_c_linear, input_axes_without_c, output_axes_without_c)
+    assert case in [1, 5]
+
+    # we need to deal with the equivalent of _adjust_bounding_box_to_real_axes() to account for the fact that the
+    # intrinsic coordinate system of the points could be ('x', 'y', 'z'). The counter image of the polygon in such
+    # coordinate system could be not orthogonal to the 'z' axis; in such a case we would need to project the points to
+    # the plan in which the polygon live. Let's not handle this complexity and simply raise an error if the inverse
+    # transformation is mixing the 'z' axis with the other axes.
```

Comment:
I'm sorry I don't understand, when would be the case when" the inverse of the transformation is mixing the z axis" ? What does mixing means here?

Reply:
This is also unclear to me. Also, what is "counter image"?

Reply:
I will rewrite the comment in a clearer way. Here "counter image" is a bad translation of the Italian "controimmagine". I checked and it's called "preimage" in English.

Reply:
Done